packages/components/src/components/Calculator/calculatorUI.tsx

```diff
@@ -99,14 +99,96 @@ export const CalculatorUI: FC<UIProps> = ({
                 {description && (
                   <div className="text-sm  text-gray-400">{description}</div>
                 )}
-                <div className="flex-grow mt-1 max-w-xs">
-                  <StyledInput
-                    value={code || ""}
-                    onChange={onChange(name)}
-                    placeholder={`Enter code for ${name}`}
-                    size="small"
-                  />
-                </div>
+                {row.tag === "text" && (
+                  <div className="flex-grow mt-1 max-w-xs">
+                    <StyledInput
+                      value={code || ""}
+                      onChange={(e) =>
+                        onChange(name)(() => {
+                          return e.target.value;
+                        })
+                      }
+                      placeholder={`Enter code for ${name}`}
+                      size="small"
+                    />
+                  </div>
+                )}
+                {row.tag === "textArea" && (
+                  <div className="flex-grow mt-1 max-w-xs">
+                    <StyledTextArea
+                      value={code || ""}
+                      onChange={(e) =>
+                        onChange(name)(() => {
+                          return e.target.value;
+                        })
+                      }
+                      placeholder={`Enter code for ${name}`}
+                    />
+                  </div>
+                )}
+                {row.tag === "checkbox" && (
+                  <div className="mt-1">
+                    <StyledCheckbox
+                      checked={(code || "true") == "true"}
+                      onChange={(e) =>
+                        onChange(name)(() => {
+                          return e.target.checked.toString();
+                        })
+                      }
+                    />
+                  </div>
+                )}
+                {row.tag === "select" && (
+                  <div className="flex-grow mt-1 max-w-xs">
+                    <Select
+                      onChange={(option) =>
+                        onChange(name)(() => {
+                          if (option) {
+                            return option.value;
+                          } else {
+                            return "";
+                          }
+                        })
+                      }
+                      value={{ value: code, label: code }}
+                      options={row.options.map((option) => ({
+                        value: option,
+                        label: option,
+                      }))}
+                      styles={{
```

Comment:
Can you give me some tips for making a StyledSelect? I'm not sure how to easily type all of the potential props (there are many).

I'm really only changing the styles, so it seems like maybe I could just export a variable for those styles, like,
```
StyledSelectStyles = {{input: ...}}
```

Reply:
The main thing the styles do is make this input consistent with other inputs; use purple instead of blue. It makes this "small", and this causes the non-centering. I'll improve this later on.